pkg/kustomize/apply.go

```diff
@@ -41,21 +46,35 @@ func (s *Kustomizer) Run(ctx context.Context, ready chan<- struct{}, stopped cha
 	kustomization := filepath.Join(s.path, "kustomization.yaml")
 	if _, err := os.Stat(kustomization); !errors.Is(err, os.ErrNotExist) {
 		logrus.Infof("Applying kustomization at " + kustomization)
-		ApplyKustomization(s.path, s.kubeconfig)
+		if err := ApplyKustomizationWithRetries(s.path, s.kubeconfig); err != nil {
+			logrus.Warnf("Applying kustomization failed: %s. Giving up.", err)
+		} else {
+			logrus.Infof("Kustomization applied successfully.")
+		}
 	} else {
 		logrus.Infof("No kustomization found at " + kustomization)
 	}
 
 	return ctx.Err()
 }
 
-func ApplyKustomization(path string, kubeconfig string) error {
+func ApplyKustomizationWithRetries(kustomization string, kubeconfig string) error {
+	err := ApplyKustomization(kustomization, kubeconfig)
+	for retry := 1; retry <= retries && err != nil; retry++ {
```

Comment:
how about reusing `wait.poll()` as in [here](https://github.com/redhat-et/microshift/blob/main/pkg/assets/crd.go#L58)

Reply:
Good idea, thank you!